docs/design/proxy.md

```diff
@@ -0,0 +1,126 @@
+    ```
+      Ex:
+        http://127.0.0.1:8888
+    ```
+  - For authenticate proxy set environment variables `VSTS_HTTP_PROXY_USERNAME` and `VSTS_HTTP_PROXY_PASSWORD` for proxy credential before start agent process.
+  - Create a `.proxybypass` file under agent root to specify proxy bypass Url's Regex (ECMAScript syntax).  
+    ```
+      Ex:
+        github\.com
+        bitbucket\.com
+    ```
+
+## How agent handle proxy within a Build/Release job
+
+After configuring proxy for agent, agent infrastructure will start talk to VSTS/TFS service through the web proxy specified in the `.proxy` file.  
+
+Since the code for `Get Source` step in build job and `Download Artifact` step in release job are also bake into agent, those steps will also follow the agent proxy configuration from `.proxy` file.  
+
+Agent will expose proxy configuration via environment variables for every task execution, task author need to use `vsts-task-lib` methods to retrieve back proxy configuration and handle proxy with their task.
+
+### Get proxy configuration by using [VSTS-Task-Lib](https://github.com/Microsoft/vsts-task-lib) method
+
+#### Node.js Lib
+
+Method for retrieve proxy settings in node.js lib
+``` typescript
+export function getHttpProxyConfiguration(): ProxyConfiguration {
+}
+```
+`ProxyConfiguration` has following fields
+```typescript
+export interface ProxyConfiguration {
+     proxyUrl: string;
+     proxyUsername?: string;
+     proxyPassword?: string;
+     proxyBypassHosts?: string[];
+ }
+```
+
+In the following example, we will retrieve proxy configuration information and use VSTS-Node-Api to make a Rest Api call back to VSTS/TFS service, the Rest call will go through the web proxy you configured in `.proxy` file.
+```typescript
+// MyProxyExampleTask.ts
+import tl = require('vsts-task-lib/task');
+import api = require('vso-node-api');
+import VsoBaseInterfaces = require('vso-node-api/interfaces/common/VsoBaseInterfaces');
+
+async function run() {
+
+    // get proxy config
+    let proxy = tl.getHttpProxyConfiguration();
+
+    // VSTS/TFS server url
+    let serverUrl = "https://myaccount.visualstudio.com";
+
+    // Personal access token
+    let token = "g6zzur6bfypfwuqdxxupv3y3qfcoudlgh26bjz77t3mgylzmvjiq";
+    let authHandler = api.getPersonalAccessTokenHandler(token);
+
+    // Options for VSTS-Node-Api
+    let option: VsoBaseInterfaces.IRequestOptions = {
+        proxy: {
+                proxyUrl: "http://127.0.0.1:8888",
+                proxyUsername: "1",
+                proxyPassword: "1",
+                proxyBypassHosts: [
+                    "github\.com"
+                ],
+            },
+            ignoreSslError: true
+        };
+    
+    // Make a Rest call to VSTS/TFS
+    let vsts: api.WebApi = new api.WebApi(serverUrl, authHandler, option);
+    let connData: lim.ConnectionData = await vsts.connect();
+    console.log('Hello ' + connData.authenticatedUser.providerDisplayName);
+}
```

Comment:
put a comment - you can use the proxy config to call other services with rest / http client (like type-rest-client or other request) by using this proxy info.